java/vector/src/test/java/org/apache/arrow/vector/TestListViewVector.java

```diff
@@ -0,0 +1,605 @@
+      resultSet = (ArrayList<ArrayList<Long>>) result;
+
+      assertEquals(3, resultSet.size()); /* 3 inner lists at index 1 */
+      assertEquals(1, resultSet.get(0).size()); /* size of the first inner list */
+      assertEquals(2, resultSet.get(1).size()); /* size of the second inner list */
+      assertEquals(3, resultSet.get(2).size()); /* size of the third inner list */
+
+      list = resultSet.get(0);
+      assertEquals(Long.valueOf(10), list.get(0));
+
+      list = resultSet.get(1);
+      assertEquals(Long.valueOf(15), list.get(0));
+      assertEquals(Long.valueOf(20), list.get(1));
+
+      list = resultSet.get(2);
+      assertEquals(Long.valueOf(25), list.get(0));
+      assertEquals(Long.valueOf(30), list.get(1));
+      assertEquals(Long.valueOf(35), list.get(2));
+
+      /* check underlying bitVector */
+      assertFalse(listViewVector.isNull(0));
+      assertFalse(listViewVector.isNull(1));
+
+      final ArrowBuf offSetBuffer = listViewVector.getOffsetBuffer();
+      final ArrowBuf sizeBuffer = listViewVector.getSizeBuffer();
+
+      // check offset buffer
+      assertEquals(0, offSetBuffer.getInt(0 * BaseRepeatedValueViewVector.OFFSET_WIDTH));
+      assertEquals(2, offSetBuffer.getInt(1 * BaseRepeatedValueViewVector.OFFSET_WIDTH));
+
+      // check size buffer
+      assertEquals(2, sizeBuffer.getInt(0 * BaseRepeatedValueViewVector.SIZE_WIDTH));
+      assertEquals(3, sizeBuffer.getInt(1 * BaseRepeatedValueViewVector.SIZE_WIDTH));
+    }
+  }
+
+  @Test
+  public void testNestedListVector1() throws Exception {
+    try (ListViewVector listViewVector = ListViewVector.empty("sourceVector", allocator)) {
+
+      MinorType listType = MinorType.LISTVIEW;
+      MinorType scalarType = MinorType.BIGINT;
+
+      listViewVector.addOrGetVector(FieldType.nullable(listType.getType()));
+
+      ListViewVector innerList1 = (ListViewVector) listViewVector.getDataVector();
+      innerList1.addOrGetVector(FieldType.nullable(listType.getType()));
+
+      ListViewVector innerList2 = (ListViewVector) innerList1.getDataVector();
+      innerList2.addOrGetVector(FieldType.nullable(listType.getType()));
+
+      ListViewVector innerList3 = (ListViewVector) innerList2.getDataVector();
+      innerList3.addOrGetVector(FieldType.nullable(listType.getType()));
+
+      ListViewVector innerList4 = (ListViewVector) innerList3.getDataVector();
+      innerList4.addOrGetVector(FieldType.nullable(listType.getType()));
+
+      ListViewVector innerList5 = (ListViewVector) innerList4.getDataVector();
+      innerList5.addOrGetVector(FieldType.nullable(listType.getType()));
+
+      ListViewVector innerList6 = (ListViewVector) innerList5.getDataVector();
+      innerList6.addOrGetVector(FieldType.nullable(scalarType.getType()));
+
+      listViewVector.setInitialCapacity(128);
+    }
+  }
+
+  private void setNewValues(ListViewVector listViewVector, List<Integer> offSets,
+      List<Integer> sizes, FieldVector elementFieldVec) {
```

Comment:
Other problems I have are: it's unusual to let you replace the child vector of a vector, and `List<Integer>` is basically the opposite of the Arrow ethos (this is a Java list of boxed integer objects, quite possibly the worst way you could represent this).

Reply:
Ah yes, that's right, better to stick to `ArrowBuf` I guess. Also we should be able to support nested lists in this approach as well?

Reply:
That's the problem...this API is complicated. Which is why I am asking for just a plain `set` (or `setUnsafe` if you insist) that does the one and only possible thing, set the validity bit + offset + length arbitrarily. It's then entirely up to the caller how they want to build the child vector.

Reply:
I think I evaluated the new API in an inaccurate and overly complicated way, it just occurred to me when I started thinking about the nested case. What you say totally makes sense to me now :100: . It is all up to the user to define the child vector.

Let me fix this and get back to you. Thanks @lidavidm

Reply:
Replaced with the suggested approach. The `setNewvalues` has been replaced with a different functionality.